plugins/field-bitmap/src/field-bitmap.js

```diff
@@ -436,6 +458,23 @@ export class FieldBitmap extends Blockly.Field {
     this.forAllCells_((r, c) => {
       newVal[r][c] = getRandBinary();
     });
+
+    /*
+    Do the same as setPixel for the randomizePixels method.
+    */
+    const oldValue = this.initialValue_;
+    if (this.getSourceBlock()) {
+      Blockly.Events.fire(
+          new (Blockly.Events.get(
+              Blockly.Events.BLOCK_FIELD_INTERMEDIATE_CHANGE))(
+              this.sourceBlock_,
+              this.name || null,
+              oldValue,
+              this.setValue(oldValue, false)
```

Comment:
This should be `newValue`. I.E. the value after we randomize the pixels.

Reply:
This still needs fixes! Here and the two unresolved comments below =)

Reply:
Hello @BeksOmega, I've pushed the updates once more. I appreciate your patience in reviewing my changes.